Lesson25/Lesson25-fercan/filter.js

```diff
@@ -0,0 +1,429 @@
+	'The butterfly gracefully fluttered its wings, showcasing vibrant colors in a delicate dance.',
+];
+
+const filteredSentences = sentencesExercise5.filter(sentence => sentence.split(' ').length <= 10);
+console.log('Ex.5:');
+console.log(filteredSentences);
+
+
+// Exercise 6:
+// Given an array of products, filter out the products that are out of stock (where the quantity is 0).
+const productsExercise6 = [
+	{
+		name: 'Laptop',
+		brand: 'Apple',
+		price: 1999,
+		quantity: 0,
+	},
+	{
+		name: 'Smartphone',
+		brand: 'Samsung',
+		price: 899,
+		quantity: 20,
+	},
+	{
+		name: 'TV',
+		brand: 'LG',
+		price: 1499,
+		quantity: 15,
+	},
+	{
+		name: 'Headphones',
+		brand: 'Sony',
+		price: 299,
+		quantity: 30,
+	},
+	{
+		name: 'Smartwatch',
+		brand: 'Apple',
+		price: 399,
+		quantity: 25,
+	},
+	{
+		name: 'Camera',
+		brand: 'Canon',
+		price: 999,
+		quantity: 0,
+	},
+	{
+		name: 'Gaming Console',
+		brand: 'Microsoft',
+		price: 499,
+		quantity: 8,
+	},
+	{
+		name: 'Tablet',
+		brand: 'Samsung',
+		price: 649,
+		quantity: 18,
+	},
+	{
+		name: 'Smart Speaker',
+		brand: 'Amazon',
+		price: 99,
+		quantity: 0,
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		quantity: 14,
+	},
+];
+
+const producstInStock = productsExercise6.filter(object => object.quantity>0);
+console.log('Ex.6:');
+console.log(producstInStock);
+
+// Exercise 7:
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+const filteredNumbers = numbersExercise7.filter(number => number % 3 !== 0);
+console.log('Ex.7:');
+console.log(filteredNumbers);
+
+// Exercise 8:
+// Given an array of strings, filter out the strings that have more than 5 characters.
+const wordsExercise8 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+console.log('Ex.8:');
+const wordsLessThan5 = wordsExercise8.filter(word => word.length<=5);
+console.log(filteredWords);
+
+
+// Exercise 9:
+// Given an array of objects representing students, filter out the students who have a grade lower than C.
+const studentsExercise9 = [
+	{
+		id: 1,
+		name: 'Alice',
+		age: 18,
+		grade: 'A',
+	},
+	{
+		id: 2,
+		name: 'Bob',
+		age: 19,
+		grade: 'B',
+	},
+	{
+		id: 3,
+		name: 'Charlie',
+		age: 20,
+		grade: 'B+',
+	},
+	{
+		id: 4,
+		name: 'David',
+		age: 19,
+		grade: 'A-',
+	},
+	{
+		id: 5,
+		name: 'Emma',
+		age: 18,
+		grade: 'A+',
+	},
+	{
+		id: 6,
+		name: 'Frank',
+		age: 20,
+		grade: 'B',
+	},
+	{
+		id: 7,
+		name: 'Grace',
+		age: 19,
+		grade: 'D+',
+	},
+	{
+		id: 8,
+		name: 'Hannah',
+		age: 20,
+		grade: 'A-',
+	},
+	{
+		id: 9,
+		name: 'Isaac',
+		age: 18,
+		grade: 'D',
+	},
+	{
+		id: 10,
+		name: 'Julia',
+		age: 19,
+		grade: 'B-',
+	},
+];
+
+console.log('Ex.9:');
+
+const studentsStudyMore = studentsExercise9.filter(student => student.grade<'C');
```

Comment:
This solution would not work as JS cannot correctly compare non-numbers using operators like '>'.

We know that grades that are lower that C are C-, all D and F grades.
The easy solution could be something like this:
studentsExercise9.filter(student => student.grade[0].toUpperCase() === 'D' || student.grade[0].toUpperCase() ==='F' || student.grade.toUpperCase() === 'C-')

In this solution we take only the first letter of D and F grades so that plus or minus symbols after the letter does not affect the result. For C- we need precise comparison so entire grade is used.